src/v/redpanda/debug_bundle.h

```diff
@@ -25,7 +26,7 @@
 
 static constexpr ss::shard_id debug_bundle_shard_id = 0;
 
-enum class debug_bundle_status { initialized, running, complete };
+enum class debug_bundle_status { initialized, running, complete, deleted };
```

Comment:
Do we need this?  After it's deleted, a fetch should return a 404 and the user knows there isn't a file present.

Reply:
> Do we need this? 

I think so. After the bundle is deleted, it does not make sense to me to set it's status to `initialized` or anything else. Also, without this status flag, I do not see how we can differentiate between a bundle that is aborted/finished-running (i.e., `completed`) and a bundle that is removed from disk (i.e. `deleted`). According to the PRD, those are two separate states.

> After it's deleted, a fetch should return a 404 and the user knows there isn't a file present.

According to the PRD, status 404 is for "When a bundle is in progress but NOT yet available to return". Status 410 is for "When a bundle is not in progress and not available to return" (i.e., rpk is not running yet or the bundle is removed from disk).

So, I will keep the deleted status.

Reply:
> According to the PRD, status 404 is for "When a bundle is in progress but NOT yet available to return" (i.e., rpk debug bundle is running but not complete yet). Status 410 is for "When a bundle is not in progress and not available to return" (i.e., rpk is not running yet or the bundle is removed from disk).

Can't this be checked by:

```
if (!bundleExists()) {
   if (!running() {
       return 410;
   } else {
      return 404;
   }
}
```

I'm not sure that we need to track a state variable as we can infer the state by checking whether or not the process is running and whether the file exists.

Reply:
I thought the same thing. Here is my rebuttal:

1. ~~`BundleExists()` would require another `process_spawn` to check that the bundle is on disk. I'd rather limit the number of posix spawns.~~ EDIT: I guess I could use C++ filesystem header here
2. Checking if the bundle exists is simply another way to check bundle state. IMO, there is no difference in semantics between your suggestion and what I have already written. I am open to being convinced otherwise though so let's talk about it ove voice chat.

Reply:
Spoke offline with @michael-redpanda:
We're going to use the filesystem to check for if a bundle is on disk or not instead of an enum. I will change the enum back to a binary status.